@here/harp-materials/lib/CirclePointsMaterial.ts

```diff
@@ -43,46 +45,47 @@ export interface CirclePointsMaterialParameters extends THREE.ShaderMaterialPara
      * Point size.
      */
     size?: number;
-}
 
-const DEFAULT_CIRCLE_SIZE = 1;
+    /**
+     * Point color.
+     */
+    color?: THREE.Color;
+}
 
 /**
  * Material designed to render circle points. Note that it is always transparent since the circle
  * shape is created with an alpha channel to benefit an antialising that a mere `discard` could
  * not bring.
  */
 export class CirclePointsMaterial extends THREE.ShaderMaterial {
-    isCirclePointsMaterial: true;
-    uniforms: { [uniform: string]: THREE.IUniform };
-    vertexShader: string;
-    fragmentShader: string;
+    static readonly DEFAULT_CIRCLE_SIZE = 1;
 
-    private m_size: number;
     private m_color: THREE.Color;
+    private m_opacity: number;
 
     /**
      * Constructs a new `CirclePointsMaterial`.
      *
      * @param parameters The constructor's parameters.
      */
     constructor(parameters: CirclePointsMaterialParameters = {}) {
-        parameters.depthTest = false;
+        const { size, color, opacity, ...shaderParams } = parameters;
+        shaderParams.name = "CirclePointsMaterial";
+        shaderParams.vertexShader = vertexShader;
+        shaderParams.fragmentShader = fragmentShader;
+        shaderParams.depthTest = false;
 
-        super(parameters);
+        super(shaderParams);
         enforceBlending(this);
```

Comment:
Please also consider using blending only when opacity is below < 1.0.

Reply:
We need top keep blending enabled even if opacity is 1.0. The circle material uses alpha blending to create a smooth edge. So the edge of the circle is always transparent.